### DISCUSSION TITLE
Phantom Ranch - Grand Canyon National Park

### DISCUSSION THREAD

**Discussion by adventurer1979** [juftin/camply#105] [Ideas]
Is there a way to add Phantom Ranch reservations calendar to Camply.  This seems to be the calendar for general availability after people win the lottery and cancel their reservation, or if they have extra inventory.  It would be a great add to the system. Here is where you access the calendar after answering a few questions. 

https://secure.phantomranchlottery.com/phantom-ranch-lottery/availability/check

**Comment by juftin** [comment]
Hey @adventurer1979 camply is currently only set up with two `"providers"` - https://recreation.gov and https://www.yellowstonenationalparklodges.com/stay/camping/

So, for now, it doesn't integrate with the Phantom Ranch lottery. My vision for camply, though, is to make it easy to add new providers with as little custom code as possible. But that vision isn't totally realized and adding new providers involves a ton of exploration of undocumented APIs.

Here's the best silver lining of all though, the Phantom Ranch Lottery is actually operated by Xanterra. Xanterra also operates https://www.yellowstonenationalparklodges.com/stay/camping/ (from my understanding they're the only entity other than recreation.gov who's currently contracted to operate inside National Parks). Hopefully this means that I can take my Yellowstone specific code and expand it to scan all campgrounds/reservation systems that Xanterra operates. 

No promises yet. It's hard to know if Phantom Ranch uses the same Xanterra APIs that Yellowstone uses - but it's worth digging into. Hopefully me, or some other good camply samaritan, will take a closer look closer at this soon.

**Comment by adventurer1979** [comment]
This would be amazing!  Thank you for the quick reply.  So appreciative of
your work so far to help others explore the national parks!  From a
monetary standpoint I’m sure people would donate / pay quite a bit for each
reservation as they are so very hard to come by for Phantom Ranch! Thank
you again!!

Jenn

On Fri, Jul 29, 2022 at 11:22 AM Justin Flannery ***@***.***>
wrote:

> Hey @adventurer1979 <https://github.com/adventurer1979> camply is
> currently only set up with two "providers" - https://recreation.gov and
> https://www.yellowstonenationalparklodges.com/stay/camping/
>
> So, for now, it doesn't integrate with the Phantom Ranch lottery. My
> vision for camply, though, is to make it easy to add new providers with as
> little custom code as possible. But that vision isn't totally realized and
> adding new providers involves a ton of exploration of undocumented APIs.
>
> Here's the best silver lining of all though, the Phantom Ranch Lottery is
> actually operated by Xanterra. Xanterra also operates
> https://www.yellowstonenationalparklodges.com/stay/camping/ (from my
> understanding they're the only entity other than recreation.gov who's
> currently contracted to operate inside National Parks). Hopefully this
> means that I can take my Yellowstone specific code and expand it to scan
> all campgrounds/reservation systems that Xanterra operates.
>
> No promises yet. It's hard to know if Phantom Ranch uses the same Xanterra
> APIs that Yellowstone uses - but it's worth digging into. Hopefully me, or
> some other good camply samaritan, will take a closer look closer at this
> soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/juftin/camply/discussions/105#discussioncomment-3282043>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2JEJ6PGH3SIIAK4LUOR45LVWQAMXANCNFSM55BGN2XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Thanks,

Jennifer Hemingway
cell: 214.417.1995
